### DISCUSSION TITLE
TemperatureDependentHardeningStressUpdate not handling temperatures out of defined temperature-hardening function range correctly.

### DISCUSSION THREAD

**Discussion by Edward-Eth** [idaholab/moose#21395] [Q&A Modules: Solid mechanics] [answered]
Looking at the source code for TemperatureDependentHardeningStressUpdate, it seems that if you have hardening functions at temperatures of 100, 200, 300, 400, and a node is at a temperature of 500, the lines: 
```
  else if (temp <= _hf_temperatures[0])
  {
    _hf_index_lo = 0;
    _hf_index_hi = _hf_index_lo;
    _hf_fraction = 0.0;
  }
  else if (temp >= _hf_temperatures.back())
  {
    _hf_index_lo = _hf_temperatures.size() - 1;
    _hf_index_hi = _hf_index_lo;
    _hf_fraction = 1.0;
  }
```
indicate that the node will use the hardening function defined for 400 k. However, when I run two models, one with functions at 100, 200, 300, 400, and one with functions at 100, 200, 300, 400, 600 but the 600 function is the same stress-strain function as the one defined at 400, the results are very different. This to me indicates something is going wrong with the handling of these out of range temperature values?

**Comment by Edward-Eth** [comment]
Will update with input file once I make one stripped back to the minimum required to show this issue.

**Comment by GiudGiud** [comment, marked as answer]
Hello

How would you rather is handles them? Erroring is not always the best option, these values could be visited only temporarily during the nonlinear solve.
Extrapolations eventually dont make sense either.
It's up to modeler to make sure their result is within bounds.

What we should do is document this behavior though. Where is the documentation for that object?

Guillaume

**Reply by Edward-Eth** [reply]
Hi, sorry I've tried recreating this issue in a simpler input file and it seems to be working, so it might have been a different issue previously. I think the current behaviour is the right thing to do, but yes it isn't actually listed in the documentation how this object handles temperatures out of the range for which hardening functions are assigned, which would be a worthwhile addition to the documentation.
